clustered/integration-test/src/test/java/org/ehcache/clustered/management/ClusterWithManagement.java

```diff
@@ -0,0 +1,130 @@
+import org.junit.runner.Description;
+import org.junit.runners.model.MultipleFailureException;
+import org.junit.runners.model.Statement;
+import org.terracotta.connection.Connection;
+import org.terracotta.connection.ConnectionException;
+import org.terracotta.exception.EntityConfigurationException;
+import org.terracotta.management.entity.nms.NmsConfig;
+import org.terracotta.management.entity.nms.client.DefaultNmsService;
+import org.terracotta.management.entity.nms.client.IllegalManagementCallException;
+import org.terracotta.management.entity.nms.client.NmsEntity;
+import org.terracotta.management.entity.nms.client.NmsEntityFactory;
+import org.terracotta.management.entity.nms.client.NmsService;
+import org.terracotta.management.model.cluster.ServerEntity;
+import org.terracotta.management.model.cluster.ServerEntityIdentifier;
+import org.terracotta.testing.rules.Cluster;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static java.lang.Thread.sleep;
+import static java.util.Arrays.asList;
+
+public final class ClusterWithManagement implements TestRule {
+
+  private final Cluster cluster;
+
+  private Connection managementConnection;
+  private NmsService nmsService;
+  private ServerEntityIdentifier tmsServerEntityIdentifier;
+
+  public ClusterWithManagement(Cluster cluster) {
+    this.cluster = cluster;
+  }
+
+  protected void before() throws Throwable {
+    this.managementConnection = cluster.newConnection();
+    this.nmsService = createNmsService(managementConnection);
+    while ((tmsServerEntityIdentifier = nmsService.readTopology()
+      .activeServerEntityStream()
+      .filter(serverEntity -> serverEntity.getType().equals(NmsConfig.ENTITY_TYPE))
+      .filter(ServerEntity::isManageable)
+      .map(ServerEntity::getServerEntityIdentifier)
+      .findFirst()
+      .orElse(null)) == null) {
+      sleep(100);
+    }
+    startCollectingServerEntityStats();
+  }
+
+  protected void after() throws Exception {
+    try {
+      nmsService.readTopology().getSingleStripe().getActiveServerEntity(tmsServerEntityIdentifier)
+        .ifPresent(client -> {
+          try {
+            nmsService.stopStatisticCollector(client.getContext());
+          } catch (Exception e) {
+            throw new RuntimeException(e);
+          }
+        });
+    } finally {
+      managementConnection.close();
+    }
+  }
+
+  @Override
+  public Statement apply(Statement base, Description description) {
+    return cluster.apply(new Statement() {
+      @Override
+      public void evaluate() throws Throwable {
+        before();
+        try {
+          base.evaluate();
+          after();
+        } catch (Throwable t) {
+          try {
+            after();
+            throw t;
+          } catch (Exception e) {
+            throw new MultipleFailureException(asList(t, e));
+          }
+        }
+      }
+    }, description);
+  }
```

Comment:
It would be cleaner to depend on that: https://github.com/Terracotta-OSS/terracotta-utilities/pull/9/files

FYI I have modified the PR to not re-use the JUnit one, that is not able to passe the method description  to after() and before(), and is also not able to throw an exception from after().

So this is a different one now.

`ExtendedTestRule` is currently copied in tc-platform in or `common-test-utilities` module and is used across all our rules.

Also, `ExtendedTestRule` with the description passed to before() is required to correctly replace the superclass of Galvan `BasicExternalCluster` class, which need to get the test description, and to avoid replacing a test failure by a galvan cleanup failure.

Anyway... All of that is cleaned in tc-platform. But it would be nice to have a common behavior.